tools/cli/src/cli/main.rs

```diff
@@ -13,13 +13,39 @@
 
 use clap::Parser;
 use kanidm_cli::KanidmClientParser;
+use std::process::ExitCode;
 use std::thread;
 use tokio::runtime;
+use tokio::signal::unix::{signal, SignalKind};
 use tracing_subscriber::filter::LevelFilter;
 use tracing_subscriber::prelude::*;
 use tracing_subscriber::{fmt, EnvFilter};
 
-fn main() {
+async fn signal_handler(opt: KanidmClientParser) -> ExitCode {
```

Comment:
Actually... this needs a Unix /windows target split

Reply:
Will do